host-controller/src/main/java/org/jboss/as/host/controller/DomainModelControllerService.java

```diff
@@ -1268,6 +1267,24 @@ public ModelNode executeReadOnly(ModelNode operation, OperationStepHandler handl
         public ModelNode executeReadOnly(ModelNode operation, Resource model, OperationStepHandler handler, OperationTransactionControl control) {
             return executeReadOnlyOperation(operation, model, control, handler);
         }
+
+        @Override
+        public void acquireReadlock(final Integer operationID) throws IllegalArgumentException, InterruptedException {
+            if (operationID == null) {
+                throw new IllegalArgumentException("operationID may not be null.");
+            }
```

Comment:
Needs i18n, or instead of the IAE an "assert operationID != null;" is probably ok. I use assert a lot for checks of internal contracts.

i18n is easy though:

```
throw HostControllerLogger.DOMAIN_LOGGER.nullVar("operationID");
```

Same comment applies to releaseReadLock.